### PR TITLE
Generate specification modules with solc-to-k and foundry-to-k

### PR DESCRIPTION
**Pull request by ehildenb**
Before, we had separate command `gen-spec` which would have to be called after `solc-to-k` and `foundry-to-k`. This was because we didn't have a good way to unparse the specifications correctly before #1290. This PR:

- Generates specification modules for `solc-to-k` and `foundry-to-k` directly in the same definition as the main modules.
- Removes the now redundant `gen-spec` command and associated tests.
- Refactors `kevm` script to allow passing `--bug-report` to `kevm run ...` with the haskell backend.
- Factors out common argumenst between `solc-to-k` and `foundry-to-k`.
- Allows passing in `--require` and `--module-import` to `solc-to-k` and `foundry-to-k` to let the user specify extra modules that should be included for verification.
- Verifies the generated specs from `foundry-to-k` on CI.

### CODE REVIEW

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 1]
```diff
@@ -98,24 +96,20 @@ def gen_claims_for_contract(empty_config: KInner, contract_name: str) -> List[KC
...
-    return [claim]
+    init_term = substitute(empty_config, init_subst)
+    init_terms = [(contract_name.lower(), init_term)]
+    if calldata_cells:
+        init_terms = [(f'{contract_name.lower()}-{i}', substitute(init_term, {'CALLDATA_CELL': cd})) for i, cd in enumerate(calldata_cells)]
+    final_term = abstract_cell_vars(substitute(empty_config, final_subst))
+    claims: List[KClaim] = []
+    for claim_id, i_term in init_terms:
+        claim, _ = build_rule(claim_id, CTerm(i_term), CTerm(final_term), claim=True)
+        assert isinstance(claim, KClaim)
```
Comment (tothtamas28) [inline kevm_pyk/src/kevm_pyk/solc_to_k.py:107]: We should have a function `build_claim` in `pyk` that wraps this logic.

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 2]
```diff
@@ -98,24 +96,20 @@ def gen_claims_for_contract(empty_config: KInner, contract_name: str) -> List[KC
...
     final_subst = {'K_CELL': KSequence([KEVM.halt(), KVariable('CONTINUATION')])}
-    init_term = CTerm(substitute(empty_config, init_subst))
-    final_term = CTerm(abstract_cell_vars(substitute(empty_config, final_subst)))
-    claim, _ = build_rule(contract_name.lower(), init_term, final_term, claim=True)
-    assert isinstance(claim, KClaim)
-    return [claim]
+    init_term = substitute(empty_config, init_subst)
+    init_terms = [(contract_name.lower(), init_term)]
+    if calldata_cells:
+        init_terms = [(f'{contract_name.lower()}-{i}', substitute(init_term, {'CALLDATA_CELL': cd})) for i, cd in enumerate(calldata_cells)]
```
Comment (tothtamas28) [inline kevm_pyk/src/kevm_pyk/solc_to_k.py:102]:
```suggestion
    if calldata_cells:
        init_terms = [(f'{contract_name.lower()}-{i}', substitute(init_term, {'CALLDATA_CELL': cd})) for i, cd in enumerate(calldata_cells)]
    else:
        init_terms = [(contract_name.lower(), init_term)]
```

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 3]
```diff
@@ -139,7 +133,22 @@ def contract_to_k(contract_json: Dict, contract_name: str, generate_storage: boo
...
     module = KFlatModule(module_name, sentences, [KImport('EDSL')])
 
-    return module
+    claims_module: Optional[KFlatModule] = None
+    function_test_productions = [prod for prod in module.syntax_productions if prod.sort == KSort(f'{contract_name}Function')]
+    contract_function_application_label = KLabel(f'function_{contract_name}')
+    function_test_calldatas = []
+    for ftp in function_test_productions:
+        klabel = ftp.klabel
+        assert isinstance(klabel, KLabel)
```
Comment (tothtamas28) [inline kevm_pyk/src/kevm_pyk/solc_to_k.py:142]:
```suggestion
        assert klabel is not None
```

**Review by tothtamas28 — APPROVED** [review]

File: kevm_pyk/src/kevm_pyk/solc_to_k.py (hunk 1 above)
Comment (ehildenb) [reply kevm_pyk/src/kevm_pyk/solc_to_k.py:107]: Added here: https://github.com/runtimeverification/ksummarize/issues/82

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 4]
```diff
@@ -139,7 +133,22 @@ def contract_to_k(contract_json: Dict, contract_name: str, generate_storage: boo
     sentences = [contract_subsort, contract_production] + storage_sentences + function_sentences + [contract_macro] + function_selector_alias_sentences
     module = KFlatModule(module_name, sentences, [KImport('EDSL')])
 
-    return module
+    claims_module: Optional[KFlatModule] = None
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply kevm_pyk/src/kevm_pyk/solc_to_k.py:136]: That's coming in a follow-on PR.

**Pull request merged**